src/integrationTest/groovy/UmplePluginIntegrationTests.groovy

```diff
@@ -0,0 +1,88 @@
+import org.junit.*
+import static org.junit.Assert.*
+import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.api.Project
+import org.gradle.tooling.* //GradleConnector, ProjectConnection, BuildLauncher
+
+public class UmpleIntegrationTests {
+    private static Project testProject
+    private static final DIRECTORIES_TO_CHECK = 
+    [
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}subproj${File.separator}build${File.separator}classes${File.separator}generatedSource",
```

Comment:
How about this?

```
private static final DIRECTORIES_ROOT = "src/integrationTest/resources/TestProject/"
    private static final DIRECTORIES_TO_CHECK = 
    [
        DIRECTORIES_ROOT + "subproj/build/classes/generatedSource",
        DIRECTORIES_ROOT + "subproj/generated/java",
        DIRECTORIES_ROOT + "sub2/build/classes/generatedSource",
        DIRECTORIES_ROOT + "sub2/generated/java"
    ] 
```

If the array just contains the subpaths for each directory we'd need to append the root before using any of the values in the array - that's easy to forget to do. We'd also be appending strings unnecessarily (rather than appending the root just once in the array we'd be appending the root and the subpath anywhere we wanted to use the array values)